hub/core/storage/memory.py

```diff
@@ -1,19 +1,86 @@
-from hub.core.storage.mapped_provider import MappedProvider
-import fsspec  # type: ignore
+from hub.core.storage.provider import StorageProvider
 
 
-class MemoryProvider(MappedProvider):
+class MemoryProvider(StorageProvider):
```

Comment:
Is it correct to say that this entire class is just a dict? I think it could be implemented in a couple lines by inheriting from dict or MutableMapping

Reply:
@benchislett that is what it was, but it's nice to inherit from StorageProvider for typing purposes

Reply:
that way i can say:

```
def func(storage: StorageProvider):
    pass
```

& storage can be of type `MemoryProvider`.

i like this

Reply:
Yup, that's exactly the reason

Reply:
Is the following not possible?
```
class MemoryProvider(StorageProvider, dict):
    pass
```

Reply:
I don't think so, both of these have conflicting methods